functions/api/transactions.js

```diff
@@ -76,17 +89,65 @@ router.get("/:id", isAdmin, async (request, response, next) => {
         transactionCollection = transactionCollection.where(constants.DATE,">=",new Date(fromDate));
     }
     if(toDate){
-        transactionCollection = transactionCollection.where(constants.DATE,"<=",new Date(toDate));
+        toDate = new Date(toDate)
+        // adding a day to the given query date as the new Date function returns 00:00:00 and we 
+        // need to include the toDate for query .
+        toDate.setDate(toDate.getDate()+1)
+        console.log(toDate);
+        transactionCollection = transactionCollection.where(constants.DATE,"<",toDate);
+    }
+    if(nextPageToken !== null && prevPageToken !== null) {
```

Comment:
q: should the check be == null with. error Message and the check seems contradicting

Reply:
the thought behind this condition is to not have next and prev together in a query as it will lead to a confusion on which page to return to the frontend.

when both are null, the code will return top results with limit of page size , mean page number 1 . Below is the interaction pattern expected. 

1. UI will send transactions/{branchId} => response with first page , and npt ,ppt 
2. If UI needs next page, they shud send transactions/{branchId}?nextpagetoken=<<>> 
3. If they need prev page , they need to send transactions/{branchId}?prevpagetoken=<<>>

expectation is to get only one of them or none at any point.

Reply:
for scenario 1, first time when UI sends request: what will be the ppt. npt makes sense. Not able to understand that part.

Reply:
ppt will be null @neerk28  , i am going to have the ppt as null if there is no previous page and npt as null in case of no next page. these will also help UI folks to get the end pages